truss/cli/train/deploy_checkpoints.py

```diff
@@ -0,0 +1,359 @@
+            )
+            # add to map of checkpoints by training job id
+            checkpoints_by_training_job_id[checkpoint.training_job_id] = (
+                checkpoint_response
+            )
+        checkpoint_response = checkpoints_by_training_job_id[checkpoint.training_job_id]
+        if checkpoint.id not in checkpoint_response:
+            raise click.UsageError(f"Checkpoint {checkpoint.id} not found.")
+        if not checkpoint.name:
+            checkpoint.name = checkpoint.id
+        if not checkpoint.lora_rank:
+            checkpoint.lora_rank = _get_lora_rank(checkpoint_response[checkpoint.id])
+    return checkpoint_details
+
+
+def _get_checkpoint_ids_to_deploy(checkpoint_id_options: List[str]) -> str:
+    if len(checkpoint_id_options) == 0:
+        raise click.UsageError("No checkpoints found for the training job.")
+    if len(checkpoint_id_options) > 1:
+        checkpoint_ids = inquirer.checkbox(
+            message="Select the checkpoint to deploy. Use spacebar to select/deselect.",
+            choices=checkpoint_id_options,
+        ).execute()
+        if not checkpoint_ids:
+            raise click.UsageError("At least one checkpoint must be selected.")
+    else:
+        checkpoint_ids = [checkpoint_id_options[0]]
+    return checkpoint_ids
+
+
+def _hydrate_checkpoints(
+    job_id: str, checkpoint_id: str, response_checkpoints: OrderedDict[str, dict]
+) -> Checkpoint:
+    if checkpoint_id == "latest":
+        checkpoint_id = list(response_checkpoints.keys())[-1]
+    checkpoint = response_checkpoints[checkpoint_id]
+    return Checkpoint(
+        training_job_id=job_id,
+        id=checkpoint_id,
+        name=checkpoint_id,
+        lora_rank=_get_lora_rank(checkpoint),
+    )
+
+
+def _get_lora_rank(checkpoint_resp: dict) -> int:
+    lora_adapter_config = checkpoint_resp.get("lora_adapter_config") or {}
+    return lora_adapter_config.get("r") or DEFAULT_LORA_RANK
+
+
+def _get_hf_secret_name(
+    console: Console, user_input: Union[str, SecretReference, None]
+) -> str:
+    if not user_input:
+        # prompt user for hf secret name
+        hf_secret_name = inquirer.select(
+            message="Enter the huggingface secret name.",
+            choices=["hf_access_token", None, "custom"],
+            default="hf_access_token",
+        ).execute()
+        if hf_secret_name == "custom":
+            hf_secret_name = inquirer.text(
+                message="Enter the huggingface secret name."
+            ).execute()
+        if not hf_secret_name:
+            console.print("No hf secret name.", style="yellow")
+        return hf_secret_name
+    if isinstance(user_input, SecretReference):
+        return user_input.name
+    return user_input
+
+
+def _get_compute(compute: Optional[Compute]) -> Compute:
+    if not compute:
+        compute = Compute()
+    compute.accelerator = _get_accelerator_if_specified(compute.accelerator)
+    if not compute.accelerator:
```

Comment:
nit: should the above fn be able to default to CPU, so this doesn't need to override?